mgtool/mgpath/path.go

```diff
@@ -0,0 +1,50 @@
+package mgpath
+
+import (
+	"bytes"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+// Path This should only be used to set a custom value.
+// Targets should use path() instead which performs
+// validation on whether a path is set.
+// nolint: gochecknoglobals
+var mgToolsPath = FromGitRoot(".mage/tools")
+
+func FromWorkDir(path string) string {
+	cwd, err := os.Getwd()
+	if err != nil {
+		panic(err)
+	}
+	return filepath.Join(cwd, path)
+}
+
+func FromGitRoot(path string) string {
+	// We use exec.command here because this command runs in a global,
+	// which is set up before mage has configured logging resulting in unwanted log prints
+	output := &bytes.Buffer{}
+	c := exec.Command("git", []string{"rev-parse", "--show-toplevel"}...)
+	c.Env = os.Environ()
+	c.Stderr = os.Stderr
+	c.Stdout = output
+	c.Stdin = os.Stdin
+
+	if err := c.Run(); err != nil {
+		panic(err)
+	}
+	return filepath.Join(strings.TrimSpace(output.String()), path)
+}
+
+func Tools() string {
+	if mgToolsPath == "" {
+		panic("No tools path set")
+	}
+	return mgToolsPath
+}
+
+func SetTools(p string) {
```

Comment:
IMO I think we should not make this configurable until there's a compelling reason to do it!